samples/typescript_nodejs/13.core-bot/src/dialogs/flightBookingRecognizer.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { RecognizerResult, TurnContext } from 'botbuilder';
+import { LuisApplication, LuisRecognizer } from 'botbuilder-ai';
+
+export class FlightBookingRecognizer {
+    private recognizer: LuisRecognizer;
+
+    constructor(config: LuisApplication) {
+        const luisIsConfigured = config && config.applicationId && config.endpoint && config.endpointKey;
+        if (luisIsConfigured) {
+            this.recognizer = new LuisRecognizer(config, {}, true);
+        }
+    }
+
+    public get isConfigured(): boolean {
+        return (this.recognizer !== undefined);
+    }
+
+    /**
+     * Returns an object with preformatted LUIS results for the bot's dialogs to consume.
+     * @param {TurnContext} context
+     */
+    public async executeLuisQuery(context: TurnContext): Promise<RecognizerResult> {
+        return await this.recognizer.recognize(context);
```

Comment:
(Optional)

This can be simplified to a non-async function that returns the promise directly:

```
public async executeLuisQuery( ... ): Promise<RecognizerResult> {
  return this.recognizer.recognize(context);
}
```

All the async keyword here is doing is wrapping the underlying promise in an extra promise.

Reply:
Good catch, thanks for pointing this out 👍

Reply:
edited the code snippet, accidentally left the `async` keyword in